spec/abci/abci++_comet_expected_behavior.md

```diff
@@ -155,9 +155,8 @@ Let us now examine the grammar line by line, providing further details.
 >consensus-round     = proposer / non-proposer
 >```
 
-* For every round, if the local process is the proposer of the current round, CometBFT calls `PrepareProposal`, followed by `ProcessProposal`.
-These two always come together because they reflect the same proposal that the process
-also delivers to itself.
+* For every round, if the local process is the proposer of the current round, CometBFT calls `PrepareProposal` and then `ProcessProposal`.
+These two always come together because once a proposal is prepared it is broadcast to all validators, including the proposer itself. In the absence of restarts, `ProcessProposal` is invoked with the proposal returned by `PrepareProposal`.
```

Comment:
Sorry, it is not this part that we have to change?

Reply:
If the node is the proposer of the current round, we call `PrepareProposal`.

If the proposer doesn't crash, `ProcessProposal` is called for the same (prepared) proposal.

If the proposer crashes and then recovers, there are some possibilities, we cannot ensure in the current implementation this match of prepare and process the same proposal.

Reply:
It was changed. Let me try to make it more clear.

Reply:
Sorry, we had a concurrency problem here probably. Now I've read you preferred not update the wording here because the grammar says otherwise.